dali/kernels/reduce/mean_gpu.cu

```diff
@@ -0,0 +1,107 @@
+// Copyright (c) 2020, NVIDIA CORPORATION. All rights reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#include <memory>
+#include "dali/kernels/reduce/reduce_gpu.h"
+#include "dali/kernels/reduce/mean_stddev_gpu_impl.cuh"
+
+namespace dali {
+namespace kernels {
+
+template <typename Out, typename In>
+class MeanGPU<Out, In>::Impl : public reduce_impl::MeanImplGPU<Out, In> {
+};
+
+template <typename Out, typename In>
```

Comment:
Do you really need to `default` constructors and destructors in the .cu file instead of inline in class definition?

Reply:
Yes, I do - otherwise unique_ptr complains about incomplete type.